### PR TITLE
Allow opening files at a specific line and column (fixes #5619)

### LINKED ISSUE #5619: [Feat]: --goto flag to open file @ line:column
## What is your suggestion?

Would love to have a `--goto` flag like in VSCode, allowing to open a file, optionally specifying a line and column number.
Currently code-server allows opening files directly via cli, but doesn't allow specifying a line and column number.

## Are you interested in submitting a PR for this?

Maybe.

### PR DESCRIPTION
**Pull request by danog**
Fixes #5619

### CODE REVIEW

**Comment by jsjoeio** [comment]
@danog do you plan to add tests? also, can you add to the PR description steps for testing this locally?

**Comment by danog** [comment]
Well it's pretty simple to test this locally, just open a workspace and run `code-server -r path/to/file.ext:line:col` from the workspace directory.
It works pretty much the same as before, the only difference is that now line numbers are supported.

**Comment by danog** [comment]
Debug builds can be tested using `node node_modules/ts-node/dist/bin.js out/node/entry.js -r path/to/file.ext:line:col`

**Comment by jsjoeio** [comment]
Thank you! I tested locally and it worked like a charm!

https://user-images.githubusercontent.com/3806031/194148046-31f069ca-0e2d-4787-a63d-3382f757a061.mov

Can you please add some tests?

File: src/node/util.ts [hunk 1]
```diff
@@ -482,6 +482,15 @@ export const isFile = async (path: string): Promise<boolean> => {
...
 
+export const isDirectory = async (path: string): Promise<boolean> => {
+  try {
+    const stat = await fs.stat(path)
+    return stat.isDirectory()
+  } catch (error) {
+    return false
+  }
+}
+
```
Comment (jsjoeio) [inline src/node/util.ts:493]: Maybe add a test for this?

**Comment by jsjoeio** [comment]
cc @code-asher - i tested locally but would love to get your thoughts too 👍🏼

**Review by code-asher — APPROVED** [review]
Looks great.  Glad it was so simple!  Adding a simple test sounds like a good idea to me as well, can pretty much just copy the `isFile` one.

**Review by jsjoeio — CHANGES_REQUESTED** [review]
Add a test and then we should be good to merge!

**Review by jsjoeio — APPROVED** [review]
Awesome! Looks good to me 👍🏼

**Comment by jsjoeio** [comment]
Thanks again for your contribution @danog! 🎉

**Comment by danog** [comment]
Thanks for merging, could I have a tag please? :)

**Comment by jsjoeio** [comment]
@danog by that do you mean tagged in the release notes? :D

**Comment by danog** [comment]
I mean a new release, I've been thinking of proposing code-server as an alternative IDE at work, and our workflow heavily relies on the feature introduced in this PR :)

**Comment by jsjoeio** [comment]
Oh no way, that's exciting! I'm going to see if I can finish https://github.com/coder/code-server/pull/5624 and possibly do https://github.com/coder/code-server/issues/5625 before the next release 🤞🏼 

In the meantime, you _should_ be able to see your changes by doing `npm install -g code-server@beta` (see [here](https://www.npmjs.com/package/code-server/v/4.7.1-beta-ba68656353edfd7a50ef4b5571322e2f0c1b653c)) if you use npm to install code-server.

**Comment by alanhg** [comment]
> Debug builds can be tested using `node node_modules/ts-node/dist/bin.js out/node/entry.js -r path/to/file.ext:line:col`

I think URL param-like r support is welcome.

**Comment by jsjoeio** [comment]
We have an open issue for that: https://github.com/coder/code-server/issues/1964

**Pull request merged**